internal/node/monitor/monitor.go

```diff
@@ -0,0 +1,152 @@
+package monitor
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/rss3-network/node/provider/arweave"
+	"github.com/rss3-network/node/provider/ethereum"
+	workerx "github.com/rss3-network/node/schema/worker"
+	"github.com/rss3-network/protocol-go/schema/network"
+	"go.uber.org/zap"
+)
+
+type CheckpointState struct {
+	BlockHeight uint64 `json:"block_height"`
+	BlockNumber uint64 `json:"block_number"`
+	EventID     uint64 `json:"event_id"`
+}
+
+func (m *Monitor) CheckWorkerStatus(ctx context.Context) error {
+	for _, w := range m.config.Node.Decentralized {
+		switch w.Network {
+		case network.Arweave:
+			// get arweave client
+			arweaveClient := m.clients[w.Network].(arweave.Client)
+
+			// get checkpoint state
+			state, err := m.getCheckpointState(ctx, w.ID(), w.Network, w.Worker.String())
+			if err != nil {
+				zap.L().Error("get checkpoint state", zap.Error(err))
+				return err
+			}
+
+			currentHeight := state.BlockHeight
+
+			latestHeight, err := arweaveClient.GetBlockHeight(ctx)
+			if err != nil {
+				zap.L().Error("get latest block height", zap.Error(err))
+				return err
+			}
+
+			if err := m.DetectUnhealthy(ctx, w.Network.String(), w.Worker.String(), currentHeight, uint64(latestHeight), 500); err != nil {
```

Comment:
use of a magic number `500` here and later line 68.
use a global variable instead.